Uchu.Core/Client/Attribute/CacheMethodAttribute.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace Uchu.Core.Client.Attribute
+{
+    public enum CacheMethod
+    {
+        /// <summary>
+        /// Stores individual entries for a given id in the cache.
+        /// </summary>
+        Entry,
+        
+        /// <summary>
+        /// Temporarily stores the entire table in memory until
+        /// it isn't used.
+        /// </summary>
+        Burst,
+        
+        /// <summary>
+        /// Permanently stores the entire table in memory. Intended
+        /// only for tables that are slow to load and have a
+        /// noticeable impact to loading, such as skills.
+        /// </summary>
+        Persistent,
+    }
+    
+    [AttributeUsage(AttributeTargets.Class)]
+    public class CacheMethodAttribute : System.Attribute
+    {
+        /// <summary>
+        /// Cache method to use for the table.
+        /// </summary>
+        public readonly CacheMethod Method;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="method"></param>
+        public CacheMethodAttribute(CacheMethod method)
+        {
+            this.Method = method;
```

Comment:
you don't have to use `this` here

Reply:
The `this` is redundant, but is it worth removing? There are a lot of other cases in the code where we use `this` when it isn't needed.

Reply:
Afaik the stylegyuide used by Uchu only uses `this` when an object passes a ref to itself

Reply:
I am not sure where we keep our style guide, or if we really should care about usage of `this`. I find it a bit more clear that it is referencing a property of an object rather than a static or local variable. And if we do care... then I have a lot of `this` to remove.

Reply:
We don't have our own style guide but use the one from MS, feel free to merge if you don't really mind but it's the style guide we generally follow.

Reply:
Moving forward, I will stop using `this.`. zaop and I may consider a reformat later for some of Wincent's old code that has a new line between most lines. `enhancement/struct-packets` probably should be merged first to automatically resolve a bunch of those.